pkg/vault/test/foundry/mutation/vault/VaultAdmin.t.sol

```diff
@@ -322,41 +323,47 @@ contract VaultAdminMutationTest is BaseVaultTest {
         vault.manualReentrancyAddLiquidityToBuffer(wrappedToken, 0, 0, address(0));
     }
 
-    function testRemoveLiquidityFromBufferWhenNotVault() public {
+    function testRemoveLiquidityFromBufferHookWhenNotVaultDelegateCall() public {
         vm.expectRevert(IVaultErrors.NotVaultDelegateCall.selector);
-        vaultAdmin.removeLiquidityFromBuffer(IERC4626(address(0)), 0, address(0));
+        VaultAdmin(address(vaultAdmin)).removeLiquidityFromBufferHook(IERC4626(address(0)), 0, address(0));
     }
 
-    function testRemoveLiquidityFromBufferWhenNotUnlocked() public {
+    function testRemoveLiquidityFromBufferHookWhenVaultIsNotSender() public {
+        vm.expectRevert(abi.encodeWithSelector(IVaultErrors.SenderIsNotVault.selector, address(this)));
+        VaultAdmin(address(vault)).removeLiquidityFromBufferHook(IERC4626(address(0)), 0, address(0));
+    }
```

Comment:
```suggestion
```
Same test as `testRemoveLiquidityFromBufferHookWhenNotVault`

Reply:
it's not :)
- `msg.sender` has to be the Vault
- current address has to be vault (i.e. be a vault delegate call, since the hook actually lives in the admin)

Reply:
In other words, this one checks `onlyVault` modifier, and the other one checks `onlyVaultDelegateCall` modifier.

Not having `onlyVault` as modifier is a critical bug, as it allows anyone to unlock the vault and call `remove` hook, spoofing the sender.

Reply:
Ah I see what you mean. Fixed now.